### DISCUSSION TITLE
[CSV Import] How to import BLOCK structured datatypes?

### DISCUSSION THREAD

**Discussion by pento-thal** [pimcore/pimcore#9096] [Q&A / Support]
I'm trying to update few object materiel attributes through a csv file.

The attribute is structured as a block datatype:
![image](https://user-images.githubusercontent.com/83424908/117853626-ee67ab80-b288-11eb-95bf-f604bf627c68.png)

This is the file I would like to use:
![image](https://user-images.githubusercontent.com/83424908/117854849-2d4a3100-b28a-11eb-989b-e528666f1e83.png)

And the result should be like:
![image](https://user-images.githubusercontent.com/83424908/117855280-9cc02080-b28a-11eb-9966-de0b1e5e4205.png)

How should I proceed with column configuration to update the object?

**Comment by weisswurstkanone** [comment]
Maybe this one would be an option for you. https://pimcore.com/docs/pimcore/6.9/User_Documentation/DataObjects/CSV_Import/Operators/PHPCode.html

**Reply by pento-thal** [reply]
@weisswurstkanone thanks for the reply.

So, you're suggesting to create a custom import rule like:

```
import column 5-6 as the first block of "Tomaia"
import column 7-8 as the second block of "Tomaia"
...
```

Do you know if there is any way to deal with it without writing any code implementation?

**Reply by weisswurstkanone** [reply]
Yes, but without code there is no option.

**Reply by pento-thal** [reply]
Ok. @weisswurstkanone If I convert the attribute management from a "block" one to a "Field-Collections" one, there is a "no-code" option able to import each Field-Collection part?
